pkg/sql/exec_util.go

```diff
@@ -1748,6 +1748,8 @@ type BackupRestoreTestingKnobs struct {
 	// execution.
 	CaptureResolvedTableDescSpans func([]roachpb.Span)
 
+	RunBeforeBackupFlow func(ctx context.Context)
```

Comment:
[nit] comment.  Although if you like my other suggestion perhaps this isn't needed.